frontend/src/app/services/scenario.ts

```diff
@@ -0,0 +1,68 @@
+// Scenarios with different gameplays (e.g. reproducing real country response)
+import {MitigationState} from './game';
+
+export const scenarios: Record<string, Scenario> = {};
+
+interface ScenarioDates {
+  rampUpStartDate: string;  // Date from we start simulation
+  rampUpEndDate: string;    // Last date to which we autoplay simulation
+  startDate: string;        // First playable date
+  endDate: string;          // Last playable date
+}
+
+export class Scenario {
+  mitigationActions: MitigationAction[] = [];
+  dates: ScenarioDates;
+
+  constructor(scenarioDates: ScenarioDates) {
+    this.dates = scenarioDates;
+  }
+
+  applyMitigationActions(mitigationState: MitigationState, date: string) {
+    Object.values(mitigationState).forEach(mitigation => mitigation.active = false);
+
+    this.mitigationActions.forEach(action => {
+      if (date >= action.begin && (!action.end || date <= action.end)) {
+        mitigationState[action.mitigationId].active = true;
+      }
+    });
+  }
+  addMitigationAction(mitigationId: string, begin: string, end?: string) {
+    this.mitigationActions.push({mitigationId, begin, end});
+  }
+}
+
+interface MitigationAction {
+  mitigationId: string;
+  begin: string;
+  end: string | undefined;
+}
+
+
+// Czech Republic at the beginning of March
+const czechiaMarch2020 = new Scenario({rampUpStartDate: '2020-02-25', rampUpEndDate: '2020-03-01', startDate: '2020-03-01', endDate: '2021-07-01'});
```

Comment:
I would propose calling these e.g. `czechiaGame`, `czechiaReal`, it took me some time to understand what's their purpose.